### PR TITLE
Remove remaining references to Fedora 23

### PR DESCRIPTION
**Pull request by danmosemsft**
I am not familiar with what the .json files do, so I'm only speculating what the correct change is here. As far as I know, the references aren't breaking anything. Please advise.

@gkhanna79 @weshaggard @dagood 

Pointed out in https://github.com/dotnet/core-setup/pull/1912

### CODE REVIEW

File: pkg/Microsoft.NETCore.Platforms/runtime.json [hunk 1]
```diff
@@ -922,18 +915,11 @@
...
         },
 
-        "fedora.23-corert": {
-            "#import": [ "fedora-corert", "fedora.23" ]
-        },
-        "fedora.23-x64-corert": {
-            "#import": [ "fedora.23-corert", "fedora-x64-corert", "fedora.23-x64" ]
-        },
-
         "fedora.24-corert": {
```
Comment (danmosemsft) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: I don't know what #import does in this context, so I'm not sure of this edit.

File: pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props [hunk 2]
```diff
@@ -42,7 +42,6 @@
       <Platform>armel</Platform>
     </OfficialBuildRID>
     <OfficialBuildRID Include="debian.8-x64" />
-    <OfficialBuildRID Include="fedora.23-x64" />
     <OfficialBuildRID Include="fedora.24-x64" />
```
Comment (danmosemsft) [inline pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props]: Should fedora 25 be here?

File: src/Native/pkg/dir.props [hunk 3]
```diff
@@ -9,7 +9,6 @@
     <WinNativePath Condition="'$(WinNativePath)' == ''">$(BinDir)Windows_NT.$(PackagePlatform).$(ConfigurationGroup)\native\</WinNativePath>
     <RHELNativePath Condition="'$(RHELNativePath)' == ''">$(LinuxNativePath)</RHELNativePath>
     <DebianNativePath Condition="'$(DebianNativePath)' == ''">$(LinuxNativePath)</DebianNativePath>
-    <Fedora23NativePath Condition="'$(Fedora23NativePath)' == ''">$(LinuxNativePath)</Fedora23NativePath>
```
Comment (danmosemsft) [inline src/Native/pkg/dir.props]: Fedora25 is not here but none of these seem to be used.

File: pkg/Microsoft.NETCore.Platforms/runtime.json (hunk 1 above)
Comment (dagood) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: @ericstj could you review this?

Comment (omajid) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: Is there a general policy on removing items from the RID graph? I have seen lots of additions to the RID graph, but never a removal.

File: pkg/Microsoft.NETCore.Platforms/runtime.json [hunk 4]
```diff
@@ -922,18 +915,11 @@
...
-            "#import": [ "fedora.23-corert", "fedora.24" ]
+            "#import": [ "fedora.24" ]
         },
         "fedora.24-x64-corert": {
-            "#import": [ "fedora.24-corert", "fedora.23-x64-corert", "fedora.24-x64" ]
```
Comment (omajid) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: IMO, fedora 24 depending on fedora 23 was kind of broken, or at least didn't match anything how deps are set up elsewhere in the RID graph. Thanks for fixing this!

File: pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props (hunk 2 above)
Comment (omajid) [inline pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props]: @ellismg would know more, but I thought this defined the officially supported platforms. If Fedora 25 is not supported, it shouldn't be here. I seem to recall reading/hearing that Fedora 25 will be supported via the portable linux build.

File: pkg/Microsoft.NETCore.Platforms/runtime.json (hunk 1 above)
Comment (weshaggard) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: I don't think we should remove these from the RID graph, as there might be others that still require this in the graph. I would keep them.

**Comment by gkhanna79** [comment]
So are RIDs forever?



Thanks,
Gaurav


-------- Original message --------

**Comment by weshaggard** [comment]
>So are RIDs forever?

Good question, but I would suspect they probably need to be otherwise we risk breaking existing packages that might use them and pull in our latest RID graph. @ericstj I'm interested to hear your thoughts on this.

File: pkg/Microsoft.NETCore.Platforms/runtime.json (hunk 1 above)
Comment (ericstj) [inline pkg/Microsoft.NETCore.Platforms/runtime.json]: Do not remove.  This is effectively public surface area.

**Comment by ericstj** [comment]
RIDs are currently the only way NuGet permits for packages to have packages or assets conditional by platform.  So long as that is the case we'll need to define the graph (or someone will) so that folks can use it.

**Comment by danmosemsft** [comment]
@weshaggard @ericstj I reverted runtime.json. Look OK now?

File: pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props (hunk 2 above)
Comment (danmosemsft) [inline pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.props]: Fedora 25 *is* supported, but I'll not do anything about it here. We will certainly see if anything was missing, and I didn't introduce this problem.

**Comment by danmosemsft** [comment]
Where is that package generated to?

**Comment by gkhanna79** [comment]
@danmosemsft `bin/<Buildtype>/Packages` in CoreFX bin output.

**Comment by danmosemsft** [comment]
OK I looked in `Microsoft.Private.CoreFx.NETCoreApp.4.4.0-preview1-25203-0.nupkg` and the runtime.json has no fedora 23 reference.

**Comment by danmosemsft** [comment]
@gkhanna79 @wtgodbe can we kick the daemon now to update core-setup right away?

**Comment by wtgodbe** [comment]
I don't believe so, but @dagood would know for sure

**Comment by danmosemsft** [comment]
I believe it's possible to just go manually queue a new build for the correct build definition, but I do'nt know the correct build definition.

Or we can just wait.

**Comment by wtgodbe** [comment]
We'd have to queue a full pipebuild, since the individual legs won't queue a maestro update - the pipebuild waits for all legs to complete, then kicks off the job that does the update. I believe the next job is scheduled for 6 PM, I can queue one off earlier (now) if desired.

**Comment by dagood** [comment]
Why do you want to auto-upgrade Core-Setup now? There isn't a CoreFX build with these changes.

Maybe I'm not understanding the question, but you can queue the "Maestro General Executor" for a repo with the right parameters to do an auto-update authored by dotnet-bot. It's not easy to figure out the parameters, though--we have https://github.com/dotnet/versions/issues/99 tracking making it easy.

**Pull request merged**